contracts/Comptroller/Diamond/facets/PolicyFacet.sol

```diff
@@ -370,6 +378,202 @@ contract PolicyFacet is IPolicyFacet, XVSRewardsHelper {
+            flashLoanData.balanceAfterTransfer[j] = vTokens[j].transferOutUnderlying(receiver, underlyingAmounts[j]);
+        }
+    }
+
+    /**
+     * @notice Phase 2: Execute operations on receiver contract
+     */
+    function _executePhase2(
+        address payable initiator,
+        address payable receiver,
+        VToken[] memory vTokens,
+        uint256[] memory underlyingAmounts,
+        uint256[] memory totalFees,
+        bytes memory param
+    ) internal returns (uint256[] memory) {
+        (bool success, uint256[] memory tokensApproved) = IFlashLoanReceiver(receiver).executeOperation(
+            vTokens,
+            underlyingAmounts,
+            totalFees,
+            initiator,
+            param
+        );
+
+        if (!success) {
+            revert ExecuteFlashLoanFailed();
+        }
+        return tokensApproved;
+    }
+
+    /**
+     * @notice Phase 3: Handles repayment based on full or partial repayment
+     * @dev If full repayment is made, transfer protocol fee to protocol share reserve and update state.
+     *      If partial repayment is made, create an ongoing debt position for the unpaid balance.
+     */
+    function _executePhase3(
+        address payable initiator,
+        address payable receiver,
+        VToken[] memory vTokens,
+        uint256[] memory underlyingAmounts,
+        FlashLoanData memory flashLoanData
+    ) internal {
+        for (uint256 k = 0; k < vTokens.length; k++) {
+            _handleFlashLoan(
+                vTokens[k],
+                initiator,
+                receiver,
+                underlyingAmounts[k],
+                flashLoanData.totalFees[k],
+                flashLoanData.protocolFees[k]
+            );
+        }
+    }
+
+    /**
+     * @notice Handles the repayment and fee logic for a flash loan.
+     * @dev Transfers the repaid amount from the receiver, checks if the full amount plus fee is repaid,
+     *      and either settles the protocol fee or creates an ongoing debt position for any unpaid balance.
+     *      Updates the protocol share reserve state if the protocol fee is transferred.
+     * @param vToken The vToken contract for the asset being flash loaned.
+     * @param initiator The address of the EOA who initiated the flash loan.
+     * @param receiver The address that received the flash loan and is repaying.
+     * @param amountRepayed The amount repaid by the receiver (principal + fee).
+     * @param totalFee The total fee charged for the flash loan.
+     * @param protocolFee The portion of the total fee allocated to the protocol.
+     */
+    function _handleFlashLoan(
+        VToken vToken,
+        address payable initiator,
+        address payable receiver,
+        uint256 amountRepayed,
+        uint256 totalFee,
+        uint256 protocolFee
+    ) internal {
+        uint256 borrowedFlashLoanAmount = vToken.flashLoanAmount();
+        // Must repay full amount + fee
```

Comment:
I think this comment is not valid anymore

Reply:
https://github.com/VenusProtocol/venus-protocol/pull/545/commits/8ed2d89f60173c908de03db73c5d99d492c939ac